lib/common/xml.c

```diff
@@ -686,12 +699,28 @@ create_xml_node(xmlNode * parent, const char *name)
 
     if (parent == NULL) {
         doc = xmlNewDoc((pcmkXmlStr) "1.0");
+        if (doc == NULL) {
+            return NULL;
+        }
+
         node = xmlNewDocRawNode(doc, NULL, (pcmkXmlStr) name, NULL);
+        if (node == NULL) {
+            xmlFreeDoc(doc);
+            return NULL;
+        }
         xmlDocSetRootElement(doc, node);
 
     } else {
         doc = getDocPtr(parent);
+        if (doc == NULL) {
+            return NULL;
+        }
+
         node = xmlNewDocRawNode(doc, NULL, (pcmkXmlStr) name, NULL);
+        if (node == NULL) {
+            xmlFreeDoc(doc);
```

Comment:
we don't want to free in this case, since parent remains valid

Reply:
Ooo, good catch.